Documentation clarification for approx feature

I finally got around to updating some of my crates from ndarray v0.11.* to ndarray v0.13. I noticed that ```allclose``` was deprecated in favor of ```abs_diff_eq```. However, I noticed in order to take advantage of it you have to make use of the ```approx``` crate within your own and then use one of their exposed macros. I'm not really a fan of this. I'd rather see the traits within approx_array module exposed to users if they have the ```approx``` feature added and available within the prelude behind a ```[cfg(feature = "approx")]``` flag.  The ```numpy array doc``` also makes it seem like the ```abs_diff_eq``` is available to outside crates but as far as I can tell that's not the case.

@bluss I should have clarified in the first post. I wasn't proposing having it exposed in the prelude but instead allowing someone to do something like ```use ndarray::approx_aaray``` to access those internal functions and traits. Also, I'm with using a separate crate for the approximate comparison behind the scenes in ```ndarray```. 

A lot of this was motivated when I was updating my code from ```ndarray v0.11``` to ```ndarray v0.13``` (should have updated this sooner but life got in the way...) and got the following warning: 

```Rust
warning: use of deprecated item 'ndarray::numeric::impl_numeric::<impl ndarray::ArrayBase<S, D>>::all_close': Use `abs_diff_eq` - it requires the `approx` crate feature
  --> tests/ori_conv_tests.rs:27:30
   |
27 |     let comp = rod_comp_ori.all_close(&rod_comp_ori2, 1e-14);
   |                              ^^^^^^^^^
   |
   = note: `#[warn(deprecated)]` on by default
```

Based on this warning I would have expected the following to work as an update:

```Rust
error[E0599]: no method named `abs_diff_eq` found for struct `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>` in the current scope
  --> tests/ori_conv_tests.rs:27:30
   |
27 |     let _comp = rod_comp_ori.abs_diff_eq(&rod_comp_ori2, 1e-14);
   |                              ^^^^^^^^^^^ method not found in `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>`
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
   |
5  | use approx::abs_diff_eq::AbsDiffEq;
   |
```
 ``` mod abs_diff_eq;``` is private so ```cargo```'s suggestion doesn't work. The only thing I've found  to work is doing the following: 

```Rust
assert_abs_diff_eq!(rod_comp_ori, rod_comp_ori2, epsilon = 1e-14);
```

I hope this kinda helps motivate what I was trying to get at with the original post. I'd say one alternative to making ```array_approx``` public  would maybe having some additional documentation added related to the ```approx``` feature.

Sorry, can either of you clarify how to properly write the example above?

```
error[E0599]: no method named `abs_diff_eq` found for struct `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>` in the current scope
  --> tests/ori_conv_tests.rs:27:30
   |
27 |     let _comp = rod_comp_ori.abs_diff_eq(&rod_comp_ori2, 1e-14);
   |                              ^^^^^^^^^^^ method not found in `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>`
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
   |
5  | use approx::abs_diff_eq::AbsDiffEq;
   |
```

I can't for the life of me work it out, and as has been said, the compiler is suggesting I import a private `approx` module.

--- a/README.rst
+++ b/README.rst
@@ -83,6 +83,14 @@ your `Cargo.toml`.
   - Enables parallel iterators, parallelized methods and ``par_azip!``.
   - Implies std
 
+- ``approx``
+
+  - Implementations of traits from version 0.4 of the [`approx`] crate.
+
+- ``approx-0_5``
+
+  - Implementations of traits from version 0.5 of the [`approx`] crate.
+
 - ``blas``
 
   - Enable transparent BLAS support for matrix multiplication.

